src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1482,13 +1482,11 @@ private function addLockSection(ArrayNodeDefinition $rootNode, callable $enableI
             ->children()
                 ->arrayNode('lock')
                     ->info('Lock configuration')
-                    ->{$enableIfStandalone('symfony/lock', Lock::class)}()
                     ->acceptAndWrap(['string'], 'resources')
+                    ->{$enableIfStandalone('symfony/lock', Lock::class)}()
                     ->beforeNormalization()
                         ->ifArray()
                         ->then(static function ($v) {
-                            $v += ['enabled' => true];
```

Comment:
we can rely on canBeEnabled/Disabled to set this default, but acceptAndWrap should then be moved earlier to have a proper ordering of beforeNormalization calls (adding [enabled => true] to wrapped strings)